src/shogun/base/SGObject.h

```diff
@@ -1160,6 +1237,7 @@ class SGObject: public std::enable_shared_from_this<SGObject>
 	{
 		auto param = this->get_parameter(BaseTag(name));
 		auto cloned = any_cast<T>(param.get_value());
+		param.get_properties().remove_property(ParameterProperties::FUNCTIONAL);
```

Comment:
This is a bit awkward, but the observers pass on properties as is, meaning the value of a function call still had a FUNCTIONAL property